.github/workflows/risc0-ethereum.yml

```diff
@@ -0,0 +1,146 @@
+  contents: read
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  RISC0_TOOLCHAIN_VERSION: v2024-02-08.1
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      test: ${{ steps.filter.outputs.test }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: risc0/paths-filter@4067d885736b84de7c414f582ac45897079b0a78
+        id: filter
+        with:
+          filters: |
+            base: &base
+              - .github/actions/**
+              - .github/workflows/risc0-ethereum.yml
+              - Cargo.toml
+              - rust-toolchain.toml
+              - Cargo.lock
+            test:
+              - *base
+              - bonsai/**
+              - risc0/**
+
+  # see: https://github.com/orgs/community/discussions/26822
+  risc0-ethereum-status-check:
+    if: always()
+    needs:
+      - test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check all job status
+        # see https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#needs-context
+        # see https://stackoverflow.com/a/67532120/4907315
+        if: ${{ contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled') }}
+        run: exit 1
+
+  test:
+    if: ${{ needs.changes.outputs.test == 'true' || github.event_name == 'workflow_dispatch'}}
+    needs: changes
+    runs-on: [self-hosted, prod, Linux, cpu, docker]
+    env:
+      RUST_BACKTRACE: full
+    steps:
+      # This is a workaround from: https://github.com/actions/checkout/issues/590#issuecomment-970586842
+      - run: "git checkout -f $(git -c user.name=x -c user.email=x@x commit-tree $(git hash-object -t tree /dev/null) < /dev/null) || :"
+      - uses: actions/checkout@v4
+        with:
+          path: 'risc0'
+      - uses: ./risc0/.github/actions/rustup
+      - uses: ./risc0/.github/actions/sccache
+        with:
+          key: Linux-default
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install python dependencies
+        run: python -m pip install --upgrade toml
+      - name: Install risczero toolchain
+        run: |
+          cargo install --force --path risc0/cargo-risczero
+          cargo risczero install --version $RISC0_TOOLCHAIN_VERSION
+        working-directory: risc0
+      - uses: risc0/foundry-toolchain@2fe7e70b520f62368a0e3c464f997df07ede420f
+        with:
+          version: nightly-09fe3e041369a816365a020f715ad6f94dbce9f2
+      - name: Checkout risc0-ethereum
+        uses: actions/checkout@v4
+        with:
+          repository: 'risc0/risc0-ethereum'
+          token: ${{ secrets.GITHUB_TOKEN }}
+          submodules: recursive
+          path: 'risc0-ethereum'
+          ref: ${{ github.event.inputs.risc0-ethereum-branch || 'main' }}
+      - name: Local patch on risc0-ethereum
+        run: python ../risc0/.github/cargo_local_patch.py Cargo.toml
```

Comment:
Can this be made relative to a "repo root" variable or similar instead of relying on the current working directory?